src/tree/SubscriptionTreeItem.ts

```diff
@@ -72,9 +72,12 @@ export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
         ];
         LocationListStep.addStep(wizardContext, promptSteps);
 
+        const executeSteps: AzureWizardExecuteStep<IPostgresServerWizardContext | ICosmosDBWizardContext>[] = [];
+        executeSteps.push(new VerifyProvidersStep(['Microsoft.DocumentDB', 'Microsoft.DBforPostgreSQL']));
```

Comment:
We don't want to register both providers, because a user can only use one at a time. You should add this step in the subwizard here, after you know which type they choose:
https://github.com/microsoft/vscode-cosmosdb/blob/main/src/tree/AzureDBAPIStep.ts#L35

Reply:
Fair enough.  Registering a provider can be a pretty expensive/long task, so we probably shouldn't register both if they don't need it.

Reply:
Could also be problems where someone only has permissions for one provider or a sovereign cloud only supports one provider, but not the other

Reply:
I realized that this code format only verifies providers in the createServer() scenario and it doesn't cover with first time users with the loadMore...() scenario. 

I am not sure how to handle this situation, as we don't want to verify provider every time the servers are loaded. Any thoughts?

Reply:
This is what I do for loadMore in functions: https://github.com/microsoft/vscode-azurefunctions/blob/main/src/tree/SubscriptionTreeItem.ts#L49

If the provider isn't registered, by definition that means they don't have any resources. So all that code does is show no resources instead of the error. I'm not sure if cosmos/postgres behaves the same way and honestly that code might not work anymore after we switched to the new sdk (I should test it). In any case, the most important piece is what you've already done, which is to register when they create the first resource.

You can mess around with providers by going here in the portal: "Subscriptions" -> "Your subscription" -> "Resource providers"

Reply:
That makes sense, for the resources to not load if the provider is not registered in the first place. 
I will test it out and see how it works on databases.